### PR TITLE
Drop support for Python 3.6

### PR DESCRIPTION
**Pull request by Cadair**
This is the master branch version of #4283 seeing how it was "broken" anyway, we might as well drop it rather than fix it.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -73,12 +73,12 @@ jobs:
 
       - linux: py38-hypothesis
 
-      - linux: py36-conda
-        name: py36_conda
+      - linux: py38-conda
+        name: py368_conda
```
Comment (nabobalis) [inline azure-pipelines.yml:77]:
```suggestion
        name: py38_conda
```

**Comment by nabobalis** [comment]
Needs a changelog.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -97,9 +97,9 @@ jobs:
...
       dependsOn:
-        - py36_test
         - py37_test
+        - py38_test
         - py38_test_online
         - py38_test_32bit
-        - py36_test_oldestdeps
-        - py36_conda
+        - py38_test_oldestdeps
+        - py38_conda
```
Comment (dstansby) [inline azure-pipelines.yml:105]: Is the whole test suite run on `py38_conda`? If so it seems a bit of a waste given there's already a set of tests run on `py38_test`.

File: sunpy/__init__.py [hunk 3]
```diff
@@ -16,26 +16,13 @@
...
 except ImportError:
     print("version.py not found, please reinstall sunpy.")
     __version__ = "unknown"
 
 
-class UnsupportedPythonError(Exception):
-    pass
-
-
-if sys.version_info < tuple(int(val) for val in __minimum_python_version__.split('.')):
```
Comment (dstansby) [inline sunpy/__init__.py:33]: Is there still a check for python version somewhere that gives a nice error?

File: azure-pipelines.yml (hunk 2 above)
Comment (Cadair) [reply azure-pipelines.yml:105]: yeah, it needs to run the whole test suite. We could, but I am not sure I want to swap the conda build to macos to remove the duplication but I would rather keep them separate.

File: sunpy/__init__.py (hunk 3 above)
Comment (Cadair) [reply sunpy/__init__.py:33]: I suppose not at runtime, I will put this back.

**Comment by Cadair** [comment]
Now also bumping our other core versions. Matplotlib, scipy and pandas are now at the major version released two years before our scheduled 2.1 release (so ~Nov/Dec). Astropy gets its regular bump, so we are now at 4.0. Numpy is bumped in line with NEP 29, but also the minimum version in astropy 4.0.

File: azure-pipelines.yml (hunk 1 above)
Comment (nabobalis) [inline azure-pipelines.yml:77]:
```suggestion
        name: py38_conda
```

**Pull request merged**